txpipe/auxiliary_maps.py

```diff
@@ -0,0 +1,200 @@
+        ("shear_tomography_catalog", HDFFile),  # for per-bin psf maps
+        ("source_maps", MapsFile),  # we copy the pixel scheme from here
+    ]
+    outputs = [
+        ("aux_maps", MapsFile),
+    ]
+
+    config_options = {
+        "chunk_rows": 100_000,
+        "sparse": True,
+        "flag_exponent_max": 8,  # flag bits go up to 2**8 by default
+        "psf_prefix": "psf_",  # prefix name for columns
+        "bright_obj_threshold": 22.0,  # The magnitude threshold for a object to be counted as bright
+        "depth_band": "i",  # Make depth maps for this band
+        "snr_threshold": 10.0,  # The S/N value to generate maps for (e.g. 5 for 5-sigma depth)
+        "snr_delta": 1.0,  # The range threshold +/- delta is used for finding objects at the boundary
+    }
+    # instead of reading from config we match the basic maps
+    def choose_pixel_scheme(self):
+        with self.open_input("source_maps", wrapper=True) as maps_file:
+            pix_info = dict(maps_file.file["maps"].attrs)
+
+        return choose_pixelization(**pix_info)
+
+    def prepare_mappers(self, pixel_scheme):
+        # We make a suite of mappers here.
+
+        # We read nbin_source because we want PSF maps per-bin
+        with self.open_input("shear_tomography_catalog") as f:
+            nbin_source = f["tomography"].attrs["nbin_source"]
+        self.config["nbin_source"] = nbin_source  # so it gets saved later
+        source_bins = list(range(nbin_source))
+
+        # For making psf_g1, psf_g2 maps, per source-bin
+        psf_mapper = Mapper(
+            pixel_scheme, [], source_bins, do_lens=False, sparse=self.config["sparse"]
+        )
+
+        # for estimating depth per lens-bin
+        depth_mapper = DepthMapperDR1(
+            pixel_scheme,
+            self.config["snr_threshold"],
+            self.config["snr_delta"],
+            sparse=self.config["sparse"],
+            comm=self.comm,
+        )
+
+        # for mapping bright star fractions, for masks
+        brobj_mapper = BrightObjectMapper(
+            pixel_scheme,
+            self.config["bright_obj_threshold"],
+            sparse=self.config["sparse"],
+            comm=self.comm,
+        )
+
+        # for mapping the density of flagged objects
+        flag_mapper = FlagMapper(
+            pixel_scheme, self.config["flag_exponent_max"], sparse=self.config["sparse"]
+        )
+
+        return psf_mapper, depth_mapper, brobj_mapper, flag_mapper
+
+    def data_iterator(self):
+        band = self.config["depth_band"]
+        psf_prefix = self.config["psf_prefix"]
+        shear_catalog_type = read_shear_catalog_type(self)
+
+        # Flag column name depends on catalog type
+        if shear_catalog_type == "metacal":
+            shear_cols = [f"{psf_prefix}g1", f"{psf_prefix}g2", "mcal_flags", "weight"]
+        else:
+            shear_cols = [f"{psf_prefix}g1", f"{psf_prefix}g2", "flags", "weight"]
+
+        # See maps.py for an explanation of this
```

Comment:
eh, where is maps.py? I can't seem to find it... and yeah I'm confused what this bit is doing.

Reply:
ooooh, found it!